### PR TITLE
[24.x] rpc: treat univalue type check error as RPC_TYPE_ERROR, not RPC_MISC_ERROR

### PR DESCRIPTION
**Pull request by maflcko**
Not sure if we want to backport this, but if we do, this is the identical commits from https://github.com/bitcoin/bitcoin/pull/25737

### CODE REVIEW

**Review by fanquake — APPROVED** [review]
ACK e68d380797918e655decb76fc11725197d6d5323

**Comment by luke-jr** [comment]
Concept ACK

File: src/wallet/rpc/spend.cpp [hunk 1]
```diff
@@ -1644,7 +1643,6 @@ RPCHelpMan walletcreatefundedpsbt()
     bool rbf{wallet.m_signal_rbf};
     const UniValue &replaceable_arg = options["replaceable"];
     if (!replaceable_arg.isNull()) {
-        RPCTypeCheckArgument(replaceable_arg, UniValue::VBOOL);
         rbf = replaceable_arg.isTrue();
```
Comment (MarcoFalke) [inline src/wallet/rpc/spend.cpp:1646]: This is a behavior change. Previously it would throw on non-bool, now it will translate it to false

Comment (luke-jr) [reply src/wallet/rpc/spend.cpp:1646]: Sounds like a bug in our UniValue fork that will affect master as well?

**Comment by luke-jr** [comment]
What if we just take the first commit, and leave the redundant checks in place for 24.x?

**Comment by MarcoFalke** [comment]
Yes, first commit looks ok

**Comment by JohhnyBTC** [comment]
windows 🙄

**Pull request closed without merging**